src/components/molecules/CheckedResourcesActionsMenu/CheckedResourcesActionsMenu.tsx

```diff
@@ -68,6 +70,18 @@ const CheckedResourcesActionsMenu: React.FC = () => {
   const isInPreviewMode = useAppSelector(isInPreviewModeSelector);
   const resourceMap = useAppSelector(state => state.main.resourceMap);
 
+  const [isApplyModalVisible, setIsApplyModalVisible] = useState(false);
+
+  const checkedResources = useMemo(
+    () => checkedResourceIds.map(resource => resourceMap[resource]).filter((r): r is K8sResource => r !== undefined),
+    [checkedResourceIds, resourceMap]
+  );
+
+  const confirmModalTitle = useMemo(
+    () => `Deploy selected resources (${checkedResourceIds.length}) to cluster [${currentContext || ''}]?`,
```

Comment:
let's use the length of `checkedResources` instead of `checkedResourceIds` so we can see if we get different lengths, although it shouldn't happen